library/ssl_srv.c

```diff
@@ -1461,14 +1198,6 @@ static int ssl_parse_client_hello( mbedtls_ssl_context *ssl )
 
     buf = ssl->in_hdr;
 
```

Comment:
This is not a blocker, but the reason why we don't use `mbedtls_ssl_read_record()` here is solely because we previously wanted to gracefully handle an SSLv2 ClientHello. Now that this is being dropped, we can use the normal record parsing routines for the ClientHello, thereby simplifying the code further.

Reply:
Excellent point! That was one of my main motivations for dropping this, but in the meantime apparently I forgot about it.

As you say, it's not a blocker for this PR (it's an internal refactoring, so it's not even a blocker for 3.0). Also, I wouldn't be entirely surprised if it turned out not to work smoothly on the first try, and since this PR already tends to be pretty big, perhaps it would be better to create an issue to track that and do it separately, perhaps even as part of MPS / 1.3 work? Would you agree?

Reply:
@mpg I have done this at some point, I don't remember when and where, though - it wasn't that tricky, luckily. But I agree, we don't have to do it now or even for 3.0, as it's entirely internal. When we switch to MPS, we'll definitely unify the record fetching - keeping this feature would have hardened the MPS integration considerably.

Reply:
I've created #4224 to track this. Based on your comment, I've provisionally labeled as size S (~2 days) but feel free to re-label to M (~ 1 week) if you think that was too optimistic.